src/bundle/osx_bundle.rs

```diff
@@ -74,6 +75,24 @@ fn copy_binary_to_bundle(bundle_directory: &Path, settings: &Settings) -> crate:
         &dest_dir.join(settings.binary_name()),
     )
 }
+trait PlistEntryFormatter {
+    fn format_plist_entry(&self) -> String;
+}
+
+impl<T: AsRef<str>> PlistEntryFormatter for T {
+    fn format_plist_entry(&self) -> String {
+        let input = self.as_ref();
+        let escaped: Cow<str> = if input.contains("&") {
+            Cow::Owned(input.replace("&", "&amp;")) // Modify only if needed
+                                                    //
+                                                    // add other necessary modifications here...
+                                                    //
+        } else {
+            Cow::Borrowed(input) // No allocation if no `&` is present
```

Comment:
We're not really avoiding an allocation, since below we still make a copy when we `escaped.to_string()`, right?  Seems like we could skip the `Cow` and just:

```
if input.contains("&") {
    return input.replace("&", "&amp;");
} else {
    return input.to_string();
}
```

In fact, since the string gets copied either way, we could just skip the conditional block entirely and just `return input.replace("&", "&amp;");`

Reply:
Yes! Thanks!